internal/message_sets.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2020-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"math"
+
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protodesc"
+	"google.golang.org/protobuf/types/descriptorpb"
+)
+
+var CanSerializeMessageSets bool
+
+func init() {
```

Comment:
Instead of an init() for a rarely used feature of protobuf, we could move this check to an exported method (with `sync.Once` if it is an expensive check?). Alternatively, it could live on the options interpreter?

Reply:
Yep, `sync.Once` sounds good. And that makes sense to also just keep it in the options package since nothing else cares about it.

Reply:
done